diff --git a/hw/misc/meson.build b/hw/misc/meson.build
index 4aeeeabad6..2542e45199 100644
--- a/hw/misc/meson.build
+++ b/hw/misc/meson.build
@@ -159,6 +159,7 @@ system_ss.add(when: 'CONFIG_XLNX_VERSAL_PMC', if_true: files(
 
 system_ss.add(when: 'CONFIG_XILINX_AXI', if_true: files('stream-fifo.c'))
 system_ss.add(when: 'CONFIG_ZYNQ', if_true: files('i2c-dev-dummy.c'))
+system_ss.add(when: 'CONFIG_ZYNQ', if_true: files('ad936x_axi.c'))
 system_ss.add(when: 'CONFIG_M24CXX', if_true: files('sodimm-spd.c'))
 system_ss.add(files('gpio-mr-mux.c', 'xlnx-serbs.c'))
 
